### ISSUE TITLE
After deleting a webhook, the GET request still succeeds for that webhook

### ISSUE THREAD

**Issue by soumyalj** [rancher/rancher#7478]
Rancher Versions:
Server:v1.4.0-rc2

Docker Version::1.12.6

**OS and where are the hosts located? (cloud, bare metal, etc):**Ubuntu, Cloud

Environment Type: (Cattle/Kubernetes/Swarm/Mesos)

Steps to Reproduce:

1. Create a few webhooks
2. Delete a webhook. It gives a ```204 No Content``` as expected
3. When we delete the same webhook again it continues to give the same response. 
It should give a response webhookid not found
Also, after deleting the webhook, the GET request for that webhook, still returns the entire webhook.

```
soumyas-MacBook-Pro:notes soumya$ curl -i -H 'Accept: application/json'  -H 'Content-Type: application/json' -X DELETE http://104.154.189.176:8080/v1-webhooks/receivers/1go7?projectId=1a5
HTTP/1.1 204 No Content
Date: Thu, 19 Jan 2017 19:49:24 GMT
Expires: Thu, 01 Jan 1970 00:00:00 GMT
Server: Jetty(9.2.11.v20150529)
Set-Cookie: PL=rancher;Path=/
X-Api-Account-Id: 1a5
X-Api-Schemas: http://104.154.189.176:8080/v1/schemas
X-Rancher-Version: v1.4.0-rc2

soumyas-MacBook-Pro:notes soumya$ curl -i -H 'Accept: application/json'  -H 'Content-Type: application/json' -X DELETE http://104.154.189.176:8080/v1-webhooks/receivers/1go7?projectId=1a5
HTTP/1.1 204 No Content
Date: Thu, 19 Jan 2017 19:53:17 GMT
Expires: Thu, 01 Jan 1970 00:00:00 GMT
Server: Jetty(9.2.11.v20150529)
Set-Cookie: PL=rancher;Path=/
X-Api-Account-Id: 1a5
X-Api-Schemas: http://104.154.189.176:8080/v1/schemas
X-Rancher-Version: v1.4.0-rc2
```
```
soumyas-MacBook-Pro:notes soumya$ curl -i -H 'Accept: application/json'  -H 'Content-Type: application/json' http://104.154.189.176:8080/v1-webhooks/receivers/1go7?projectId=1a5
HTTP/1.1 200 OK
Content-Type: application/json
Date: Thu, 19 Jan 2017 19:54:03 GMT
Expires: Thu, 01 Jan 1970 00:00:00 GMT
Server: Jetty(9.2.11.v20150529)
Set-Cookie: PL=rancher;Path=/
Vary: Accept-Encoding, User-Agent
X-Api-Account-Id: 1a5
X-Api-Schemas: http://104.154.189.176:8080/v1-webhooks/schemas
X-Rancher-Version: v1.4.0-rc2
Content-Length: 953

{"actions":{},"driver":"scaleService","id":"1go7","links":{"self":"http://104.154.189.176:8080/v1-webhooks/receivers/1go7?projectId=1a5"},"name":"scaleservicesup1","scaleServiceConfig":{"action":"up","amount":1.9,"max":4.5,"min":3.5,"serviceId":"1s6"},"state":"removed","type":"receiver","url":"http://104.154.189.176:8080/v1-webhooks/endpoint?token=eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJjb25maWciOnsic2VydmljZUlkIjoiMXM2IiwiYW1vdW50IjoxLjksImFjdGlvbiI6InVwIiwibWluIjozLjUsIm1heCI6NC41fSwiZHJpdmVyIjoic2NhbGVTZXJ2aWNlIiwicHJvamVjdElkIjoiMWE1IiwidXVpZCI6ImI4ZDcxNjIzLTk3ODItNDgxMC1iYTA3LTQxYTg2YmMzNzY4ZCJ9.TSRz4MhGod53NQ8w-Z84P8uDyCj4ZpSTzJT9yqeNFSwzoK2I9KHnvavAVotRjMFjiUxmtiqP83PihsiyytlUSvAMk69Cb0fyA0WiMPllPzu23U06Pee9DSIlH3Mj0qB4IK8c0p06iUaZZWPBnPcHePL6UEfjLeG-97oVPESRcjncOEli5-YtZR3yufM1Hx45vtoAkXVhgjpVdSZH_wfvfVty8_8DmtR-myyO-mjOGbvQCJHvFce96vWg7li5A_37H33ktquGwZf9JeaAP_E16iXIo7hlgHepHI44wEITdmZApIENcLvZS3Pkd-dK3Xy3kq7p6HeQoi31kND0NzKyyw"}



```

**Comment by mrajashree** [comment]
This behavior of GET is consistent with other cattle API calls, example GET a deleted container. DELETE on a container returns 200, should webhook-service be changed to return 200 and the deleted webhook in place of 204? @cjellick

**Comment by vincent99** [comment]
204 is correct since the DELETE isn't returning any content.  Not required to release but it would be better if it actually returned 200 and a copy of the resource showing `state: removed` (or `removing`) like Cattle does.

Returning deleted resources when specifically requested by ID as long as they still exist in the database is also correct and consistent with Cattle.  The records should eventually get *actually* deleted by cleanup scripts (does this happen automatically or need to be setup @cjellick ?).  At that point asking for it by ID should return 404 just like any other ID there is no record for.

**Comment by mrajashree** [comment]
@vincent99 yes so when a deleted webhook is requested by ID, webhook-service is returning that. But if I delete it from the db manually and request it again, it gets 404.

**Comment by mrajashree** [comment]
@cjellick  since this behavior of GET and DELETE is expected, I'm reassigning this to @soumyalj without any code changes/fixes

**Comment by soumyalj** [comment]
No changes done as its an expected behavior. Hence closing.
